src/Microsoft.OData.Edm/Csdl/Parsing/SchemaJsonParser.cs

```diff
@@ -0,0 +1,1564 @@
+
+                // the member value is an object containing annotations for that target.
+                if (propertyValue.ValueKind == JsonValueKind.Object)
+                {
+                    IList<CsdlAnnotation> subAnnotations = new List<CsdlAnnotation>();
+                    propertyValue.ParseAsObject(context, (subName, subValue) =>
+                    {
+                        if (AnnotationJsonParser.TryParseCsdlAnnotation(subName, subValue, context, out CsdlAnnotation subAnnotation))
+                        {
+                            subAnnotations.Add(subAnnotation);
+                        }
+                    });
+
+                    CsdlAnnotations annotations = new CsdlAnnotations(subAnnotations, target, qualifier);
+                    annotationsCollection.Add(annotations);
+                }
+                else
+                {
+                    ReportUnexpectedElement(propertyValue, context);
+                }
+            });
+
+            return annotationsCollection;
+        }
+        #endregion
+
+        #region EntityContainer
+        /// <summary>
+        /// Parse the input <see cref="JsonElement"/> to the <see cref="CsdlEntityContainer"/>.
+        /// </summary>
+        /// <param name="name">The entity container unqualified name.</param>
+        /// <param name="jsonValue">The json value to parse from.</param>
+        /// <param name="jsonPath">The input json value path.</param>
+        /// <returns>null or the parsed <see cref="CsdlEntityContainer"/>.</returns>
+        internal static CsdlEntityContainer ParseCsdlEntityContainer(string name, JsonElement containerObject, JsonParserContext context)
+        {
+            // An entity container is represented as a member of the schema object:
+            // whose name is the unqualified name of the entity container
+            // whose value is an object.
+            Debug.Assert(containerObject.ValueKind == JsonValueKind.Object);
+
+            // The entity container object MUST contain the member $Kind with a string value of EntityContainer.
+            ValidateKind(containerObject, context, "EntityContainer", required: true);
+
+            IList<CsdlElement> entityContainerMembers = new List<CsdlElement>();
+            string extends = null;
+            containerObject.ParseAsObject(context, (propertyName, propertyValue) =>
+            {
+                switch (propertyName)
+                {
+                    case "$Kind":
+                        // Skip it, it's parsed above.
+                        break;
+
+                    case "$Extends":
+                        // The entity container object MAY contain the member $Extends,
+                        // The value of $Extends is the qualified name of the entity container to be extended.
+                        extends = propertyValue.ParseAsString(context);

```

Comment:
>extends [](start = 24, length = 7)

Probably not something we can fix until 8.x, but I would think that, if one EntityContainer extends another, then when we query the members of the entity container we should see the members of the entity container that it extends.

Reply:
filed as issue at [https://github.com/OData/odata.net/issues/1820](https://github.com/OData/odata.net/issues/1820) to track on it.

---
In reply to: [443056246](https://github.com/OData/odata.net/pull/1775#discussion_r443056246) [](ancestors = 443056246)

Reply:
perfect; thanks.

---
In reply to: [444543831](https://github.com/OData/odata.net/pull/1775#discussion_r444543831) [](ancestors = 444543831,443056246)